src/core/server/elasticsearch/is_valid_connection.test.ts

```diff
@@ -44,13 +44,28 @@ describe('isValidConnection', () => {
     await expect(promise).resolves.toStrictEqual(compatible);
   });
 
-  test('should throw an error on ProductCheckError', async () => {
+  test('should throw an error only on ProductCheckError', async () => {
     const esNodesCompatibility$ = new Subject<NodesVersionCompatibility>();
     const promise = isValidConnection(esNodesCompatibility$);
 
+    const { ProductNotSupportedError, ...otherErrors } = errors;
+
+    // Emit every declared Error by the ES client.
+    Object.entries(otherErrors).forEach(([errorName, ESError]) => {
```

Comment:
I don't think we should test *all* the errors to avoid flakiness. A single one is enough IMO

Reply:
changed to only emit 2 known errors :)